dask/dataframe/tests/test_multi.py

```diff
@@ -681,6 +681,18 @@ def test_cheap_single_partition_merge():
             a.merge(b, on='x', how='inner'))
 
 
+def test_cheap_single_partition_merge_divisions():
+    a = pd.DataFrame({'x': [1, 2, 3, 4, 5, 6], 'y': list('abdabd')},
+                     index=[10, 20, 30, 40, 50, 60])
+    aa = dd.from_pandas(a, npartitions=3)
+
+    b = pd.DataFrame({'x': [1, 2, 3, 4], 'z': list('abda')})
+    bb = dd.from_pandas(b, npartitions=1, sort=False)
+
+    assert_divisions(aa.merge(bb, on='x', how='inner'))
+    assert_divisions(bb.merge(aa, on='x', how='inner'))
```

Comment:
In my current implementation, all division information is lost. I agree, that in case of on-index joins, the division information is potentially lost unnecessarily. I will update the PR by adding corresponding tests and the required logic.